conclave-enclave/src/main/kotlin/com/r3/conclave/enclave/internal/GramineDirectEnclaveEnvironment.kt

```diff
@@ -31,14 +35,24 @@ class GramineDirectEnclaveEnvironment(
         versionToCpuSvn(tcbLevel)
     }
 
-    // TODO: (CON-1194) mrenclave value based on enclave jar content
+    /** Generate simulated mrenclave value by hashing the enclave fat-jar */
     private val mrenclave: ByteArray by lazy {
-        digest("SHA-256") { update(enclaveClass.name.toByteArray()) }
-    }
+        val jarPath = enclaveClass.protectionDomain.codeSource.location.toURI().path
+
+        val buffer = ByteArray(4096)
+        var bytesRead: Int
+
+        val digest = MessageDigest.getInstance("SHA-256")
+
+        File(jarPath).inputStream().use {
+            bytesRead = it.read(buffer)
+            while (bytesRead >= 0) {
+                digest.update(buffer, 0, bytesRead)
```

Comment:
Dumb question, it looks fine to me, but wanted to double check my understanding.
Are we updating the digest many times (and I assume "updating" does not mean"replacing" or "overriding") by providing values from the buffer from 0 to 4096 (apart from the last read).
This is how it should work, right?
But is `4096` a conventional number for buffers or what? (i.e. maybe let's try to reduce the number of iterations?)

Reply:
Yup, pretty much. "updating" here just means feeding more data into the hash function. I think this is pretty standard I think (see `java.security.MessageDigest`). As for 4096, there's no particular reason, I just didn't want to read the whole file into a byte array, and I like powers of two!

Reply:
I guess it could do to be a bit larger. What was the file block hashing size used by gramine again?

Reply:
I guess this will "update" hundreds, if not thousands, times. Not really sure about "block hashing size".
But if we don't notice immediate performance issues, we can postpone the matter and leave this as it is

Reply:
I really don't think it's a problem, this only happens once on startup. I've bumped it to 64k for now.